modules/plot/main/coffee/Utils.coffee

```diff
@@ -1,3 +1,19 @@
+doCollisionDetection = (nodes) ->
+  distance = 1
+  nodes.forEach (node, i) ->
```

Comment:
Is there a way to make this more functional? (e.g. `distance = resultOfFunction` instead of defining it outside the loop and updating)

Maybe something like this (I haven't tested it at all)
```coffee
doCollisionDetection = (nodes) ->
  distance = hx.max nodes.map (node, nodeIndex) ->
    currBox = node.getBoundingClientRect()
    previousNodes = nodes.slice 0, (nodeIndex - distance + 1)
    hx.max previousNodes.map (previousNode, prevIndex) ->
      prevBox = previousNode.getBoundingClientRect()
      doesCollide = currBox.left < prevBox.right
      if doesCollide then nodeIndex - prevIndex else 1
  nodes.forEach (node, i) ->
    if i % (distance + 1)
      hx.select(node).text ''
```